R/dvhydrograph-render.R

```diff
@@ -3,199 +3,291 @@ dvhydrographPlot <- function(data) {
-    
-    # patch up top extent of y-axis
-    plot_object <- RescaleYTop(plot_object)
-
-    #Legend
-    legend_items <- plot_object$legend$legend.auto$legend
-    ncol <- ifelse(length(legend_items) > 3, 2, 1)
-    leg_lines <- ifelse(ncol==2, ceiling((length(legend_items) - 6)/2), 0) 
-    legend_offset <- ifelse(ncol==2, 0.3+(0.05*leg_lines), 0.3)
-    plot_object <- legend(plot_object, location="below", cex=0.8, legend_offset=0.2, y.intersp=1.5, ncol=ncol)
-
-    #Add Min/Max labels if we aren't plotting min and max
-    minmax_labels <- append(dvData['max_iv_label'], dvData['min_iv_label'])
-    line <- 0.33
-    for(ml in na.omit(names(minmax_labels))){
-      #Extract Timezone
-      tzf <- format(as.POSIXct(dvData[[ml]]$time), "%z")
-      #Insert ":" before 2nd to last character
-      tzf <- sub("([[:digit:]]{2,2})$", ":\\1", tzf)
-      formatted_label <- paste0(dvData[[ml]]$legend.name, data$firstDownChain$units, 
-                                format(as.POSIXct(dvData[[ml]]$time), " %b %d, %Y %H:%M:%S"), " (UTC ", tzf, ")")
-      
-      plot_object <- mtext(plot_object, formatted_label, side = 3, axes=FALSE, cex=0.85, line = line, adj = 0)
-      
-      line <- line + 1
+    dvConfig <- getDVHydrographPlotConfig(dvData[i], yLabel=yLabel)
+    for (j in names(dvConfig)) {
+      dvConfig[[j]] <- extendStep(dvConfig[[j]])
+      plot_object <- do.call(names(dvConfig[j]), append(list(object = plot_object), dvConfig[[j]]))
     }
-    
-    return(plot_object)
   }
-  else {
-    plot_object <- NULL
-  }
-}
 
-createRefPlot <- function(data, series, descriptions) {
+  # approval bar styles are applied last, because it makes it easier to align
+  # them with the top of the x-axis line
+  plot_object <- ApplyApprovalBarStyles(plot_object, dvData)
   
-  # capitalize the reference series name for plot titles
-  ref_name_letters <- strsplit(series, "")[[1]]
-  ref_name_letters[1] <- LETTERS[which(letters == ref_name_letters[1])]
-  ref_name_capital <- paste0(ref_name_letters, collapse = "")
-    
-  if (!length(data[[series]]$points)==0) {
-    
-    refData <- parseRefData(data, series, descriptions)
-    refData <- refData$refData
-    refInfo <- refData$refInfo
-    isInverted <- fetchReportMetadataField(data, 'isInverted')
-   
-    
-    startDate <- flexibleTimeParse(data$reportMetadata$startDate, timezone=data$reportMetadata$timezone)
-    endDate <- toEndOfDay(flexibleTimeParse(data$reportMetadata$endDate, timezone=data$reportMetadata$timezone))
+  plot_object <- rmDuplicateLegendItems(plot_object)
+  
+  # custom gridlines below approval bar
+  plot_object <- plot_object %>% 
+    abline(v=seq(from=toStartOfDay(startDate), to=toStartOfDay(endDate), by="days"), lty=3, col="gray", where='first') %>%
+    abline(v=seq(from=toStartOfDay(startDate), to=toStartOfDay(endDate), by="weeks"), col="darkgray", lwd=1, where='first')
+  
+  # patch up top extent of y-axis
```

Comment:
more explicit comment  - what does "patch up" mean? If I looked at this two months from now, I don't think I'd know the point of this.